tools/@aws-cdk/spec2cdk/test/services.test.ts

```diff
@@ -25,3 +25,106 @@ test('can codegen service with arbitrary suffix', () => {
   expect(rendered).toContain('function convertCfnApplicationV2PropsToCloudFormation');
   expect(rendered).toContain('function CfnApplicationV2ApplicationCodeConfigurationPropertyValidator');
 });
+
+test('resource interface when primaryIdentifier is an attribute', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-voiceid').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnDomain` and `Domain`.',
+    ' *',
+    ' * @struct',
+    ' * @stability external',
+    ' */',
+    'export interface ICfnDomain {',
+    '  /**',
+    '   * The identifier of the domain.',
+    '   *',
+    '   * @cloudformationAttribute DomainId',
+    '   */',
+    '  readonly attrDomainId: string;',
+    '}',
+  ].join('\n'));
+});
+
+test('resource interface when primaryIdentifier is a property', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-kinesisanalyticsv2').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnApplication` and `Application`.',
+    ' *',
+    ' * @struct',
+    ' * @stability external',
+    ' */',
+    'export interface ICfnApplication {',
+    '  /**',
+    '   * The name of the application.',
+    '   *',
+    '   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kinesisanalyticsv2-application.html#cfn-kinesisanalyticsv2-application-applicationname',
+    '   */',
+    '  readonly attrApplicationName?: string;', // optional?
+    '}',
+  ].join('\n'));
+});
+
+test('resource interface with multiple primaryIdentifiers', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-lakeformation').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnDataCellsFilter` and `DataCellsFilter`.',
+    ' *',
+    ' * @struct',
+    ' * @stability external',
+    ' */',
+    'export interface ICfnDataCellsFilter {',
+    '  /**',
+    '   * Catalog id string, not less than 1 or more than 255 bytes long, matching the [single-line string pattern](https://docs.aws.amazon.com/lake-formation/latest/dg/aws-lake-formation-api-aws-lake-formation-api-common.html) .',
+    '   *',
+    '   * The ID of the catalog to which the table belongs.',
+    '   *',
+    '   * @cloudformationRef tableCatalogId',
+    '   *',
+    '   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lakeformation-datacellsfilter.html#cfn-lakeformation-datacellsfilter-tablecatalogid',
+    '   */',
+    '  readonly attrTableCatalogId: string;',
+    // '  /**',
```

Comment:
Inexplicably this line fails the comparison even though it definitely is exactly right...

Reply:
It clearly isn't 😝 

I'd probably prefer these assertions to be less focused on the exact format and more on the content. Currently you really are doing snapshot testing while trying to avoid using snapshots - and making it even harder.